### ISSUE TITLE
Why was ed25519 v1.5.2 yanked?

### ISSUE THREAD

**Issue by cbeck88** [RustCrypto/signatures#673]
Hi, I'm following up on a build error that was reported to us:

```
warning[yanked]: detected yanked crate
    ┌─ /github/workspace/Cargo.lock:123:1
    │
123 │ ed25519 1.5.2 registry+https://github.com/rust-lang/crates.io-index
    │ ------------------------------------------------------------------- yanked version
    │
    = ed25519 v1.5.2
      ├── ed25519-dalek v1.0.1
```

https://github.com/mobilecoinofficial/deqs/actions/runs/4348477341/jobs/7597062019

We get this error because we have a `deny.toml` file which detects yanked crates and reports them to us so that we can learn about security issues immediately and fix them.

It looks to me that the only difference between 1.5.2 and 1.5.3 is this change: https://github.com/RustCrypto/signatures/commit/35ee637ade5672c7c521cc56c839ca5264244c2c

---

As politely as I can, I want to suggest that this might not have been an inappropriate use of yanking. Yanking the old version just because you published a new one is harmful to downstream users.

This is not a security issue, there is nothing that urgently requires us to update from 1.5.2 to 1.5.3. To accomplish that we would have to coordinate with dalek maintainers as well, or fork dalek, which is a lot of work.

Cargo will automatically cause any new users of your crate to select 1.5.3 and not 1.5.2 due to cargo's interpretation of semver anyways.

Use of yanking when there is not a security issue will force us to disable tools that help us detect yanked crates and find actual security problems, or force us to vendor the software to ourselves outside of crates.io to avoid build breakage when checking out old commits, for example, when using git bisect to find and fix bugs.

**Comment by tarcieri** [comment]
35ee637ade5672c7c521cc56c839ca5264244c2c the important change. See also #617.

All of the yanked versions contain a *broken* version requirement:

```toml
signature = { version = ">=1.3.1", default-features = false }
```toml

This was changed to:

```toml
signature = { version = "1.3.1", default-features = false }
```

The important thing being the old version requirement also matched `signature` v2 releases, which incorporate SemVer breaking changes. The change was introduced in v1.3.0, and fixed in v1.5.3, the fix being to use proper SemVer rules.

Without it, as soon as `signature` v2.0.0 was published, all of these versions *no longer build*. The version requirement matches `signature` v2 releases which take precedence over v1 releases, and their code no longer compiles due to the breaking changes between v1 and v2.

**Comment by cbeck88** [comment]
I see, thank you
